.github/workflows/swift-toolchain.yml

```diff
@@ -876,17 +880,19 @@ jobs:
           show-progress: false
 
       - name: Install Python ${{ env.PYTHON_VERSION }} (Host)
+        if: matrix.arch == 'amd64' || inputs.build_arch == 'amd64'
         uses: actions/setup-python@v5
         id: python
         with:
           python-version: '${{ env.PYTHON_VERSION }}'
+          architecture: x64
 
       - uses: nuget/setup-nuget@v2
-        if: matrix.arch == 'arm64'
+        if: matrix.arch == 'arm64' || inputs.build_arch == 'arm64'
```

Comment:
Likewise, because we are only doing native compilation, and this is required for ARM64 only, perhaps we can leave this as `matrix.arch == 'arm64'`.

Reply:
I think the second condition is necessary because there's still the matrix case for building win/arm64 on win/x64 (in which case x64 python in the x64 build machine won't be installed) and we need python for both the build and the host arches. I think that's implied by the fact that this step's name has with "(host)" at the end.

Reply:
Oh, I see so that this workflow remains usable for cross-compilation?

Reply:
Yes!